test/common/network/address_impl_test.cc

```diff
@@ -119,6 +120,22 @@ TEST(Ipv4CompatAddressImplSocktTest, SocketBindAndConnect) {
   }
 }
 
+TEST(Ipv4InstanceTest, SockaddrToString) {
+  // Test addresses from various RF 5735 reserved ranges
+  static const char* addresses[] = {"0.0.0.0",        "0.0.0.255",       "0.0.255.255",
+                                    "0.255.255.255",  "192.0.2.0",       "198.151.100.1",
+                                    "198.151.100.10", "198.151.100.100", "10.0.0.1",
+                                    "10.0.20.1",      "10.3.201.1"};
+
+  for (const auto address : addresses) {
+    sockaddr_in addr4;
+    addr4.sin_family = AF_INET;
+    EXPECT_EQ(1, inet_pton(AF_INET, address, &addr4.sin_addr));
+    addr4.sin_port = 0;
+    EXPECT_EQ(address, Ipv4Instance::sockaddrToString(addr4));
```

Comment:
EXPECT_STREQ

Reply:
The existing string checks in address_impl_test.cc all use EXPECT_EQ. What's different about this one that needs EXPECT_STREQ?

Reply:
It's mostly about legibility of output when there is a failure.  Some forms of comparing char*, char[], string_view, string, etc, result in mismatch displays showing arrays of integers rather than strings.

Reply:
I just checked, and in this case EXPECT_EQ shows both the left and right sides as strings if they don't match.

Reply:
OK usually I just recommend using EXPECT_STREQ when comparing all strings, as it's presence never hurts, and its absence can. I also thought maybe a char* to char* comparison might be a pointer-comparison if you're not careful, but obviously you'd know that pretty quick.